src/main/java6/net/finmath/marketdata/model/curves/ForwardCurveFromDiscountCurve.java

```diff
@@ -18,119 +18,110 @@
 	 * 
 	 * @param name The name under which the forward curve can be referenced.
-	 * @param discountCurveName The discount curve used for calculation of the forward.
+	 * @param referenceDiscountCurveName The (pseudo-)discount curve that the forwards are calculated from.
 	 * @param referenceDate The reference date used in the interpretation of times (i.e., the referenceDate where t=0).
 	 * @param paymentOffsetCode The payment offset. If null, the parameter p has to be provided to the getForward method.
 	 */
-	public ForwardCurveFromDiscountCurve(String name, String discountCurveName, LocalDate referenceDate, String paymentOffsetCode) {
-		super(name, referenceDate,
-				paymentOffsetCode, new BusinessdayCalendarExcludingWeekends(), BusinessdayCalendarInterface.DateRollConvention.FOLLOWING, discountCurveName
-				);
-		daycountScaling = 1.0;
-		periodOffset = 0;
+	public ForwardCurveFromDiscountCurve(String name, String referenceDiscountCurveName, LocalDate referenceDate, String paymentOffsetCode) {
+		this(name, referenceDiscountCurveName, referenceDate, paymentOffsetCode, new BusinessdayCalendarExcludingWeekends(), BusinessdayCalendarInterface.DateRollConvention.FOLLOWING);
 	}
 
 	/**
-	 * Create a forward curve using a given discount curve.
-	 * The forward with fixing in t is calculated as ( df(t)/df(t+p)-1 ) / p
-	 * where df denotes the discount factor as a function of maturity and
-	 * p is a given the payment offset.
+	 * Create a forward curve using a given referenceDiscountCurveForForwards. 
+	 * Note that the referenceDiscountCurveForForwards is also used as the discount curve associated with the forwards (i.e. single curve).
+	 * referenceDiscountCurveName + "_asForwardCurve" is used as the name for this ForwardCurveFromDiscountCurve.
 	 * 
-	 * @param discountCurveName The discount curve used for calculation of the forward.
+	 * @param referenceDiscountCurveName The (pseudo-)discount curve that the forwards are calculated from.
 	 * @param referenceDate The reference date used in the interpretation of times (i.e., the referenceDate where t=0).
 	 * @param paymentOffsetCode The payment offset. If null, the parameter p has to be provided to the getForward method.
 	 */
-	public ForwardCurveFromDiscountCurve(String discountCurveName, LocalDate referenceDate, String paymentOffsetCode) {
-		super("ForwardCurveFromDiscountCurve(" +  discountCurveName + "," + paymentOffsetCode + ")", referenceDate,
-				paymentOffsetCode, new BusinessdayCalendarExcludingWeekends(), BusinessdayCalendarInterface.DateRollConvention.FOLLOWING, discountCurveName
-				);
-		daycountScaling = 1.0;
-		periodOffset = 0;
+	public ForwardCurveFromDiscountCurve(String referenceDiscountCurveName, LocalDate referenceDate, String paymentOffsetCode) {
+		this(referenceDiscountCurveName + "_asForwardCurve", referenceDiscountCurveName, referenceDate, paymentOffsetCode);

```

Comment:
There is an issue with this change. I realised it after merging, because Jenkins UnitTests were failing ( http://ci.finmath.net/job/finmath%20lib%20-%20java%206%20-%20github%20-%20maven%20build/200/ ). The issue here is the change in how the name of the forward curve is constructed. I would like to revert to the previous method where the name is constructed via

`"ForwardCurveFromDiscountCurve(" +  discountCurveName + "," + paymentOffsetCode + ")"`

compared to the method introduced in this pull request:

`referenceDiscountCurveName + "_asForwardCurve"`

Example: Consider the case where you have one referenceDiscountCurve "discount-EUR" and you like to create two forward curves from it: one 3M and one 6M. The two forward curves will have the same name. Hence calling "model.addCurve" will store only one of them.

I don't see why the paymentOffsetCode was removed from the name.

In a future version it would be even better to make the name unique by adding a hash which also includes referenceDate, etc... but currently I don't know of any application where this is required.